src/firefly/js/charts/ChartUtil.js

```diff
@@ -519,6 +530,46 @@ export function setupTableWatcher(chartId, ts, idx) {
         uniqueId(`ucd-${ts.tbl_id}-trace`)); // watcher id for debugging
 }
 
+/**
+ * Get feedback as boolean on whether chart is fully loaded or not
+ * @param chartId - chartId of the current chart
+ */
+export function isChartLoaded(chartId) {
+    const {fireflyData=[]} = getChartData(chartId);
+    const isLoading = fireflyData.every((e)=>get(e, 'isLoading'));
+    return !isLoading; //true when chart is fully loaded (isLoading: false)
+}
+
+const onChartLoad = (action, accept, cancelSelf) => {
+    const {chartId} = action.payload;
+    if (accept(action)) {
+        if (isChartLoaded(chartId)) {
+            pinChart({chartId});
+            cancelSelf && cancelSelf(); //ActionWatcher cleanup
+        }
+    }
+    else { //if we possibly get a different tbl_id/chartId - cancel this ActionWatcher
+        cancelSelf && cancelSelf();
+    }
+};
+
+
+export function setupChartWatcher(tbl_id) {
+    const accept = (a) => 'default-'+tbl_id === (get(a, 'payload.chartId') || get(a, 'payload.request.chartId'));
+    return watchChartChanges([CHART_UPDATE],
+        (action, cancelSelf) => onChartLoad(action, accept, cancelSelf));
+}
+
+/**
+ * this function invokes the given callback when changes are made to a Chart (in this case, CHART_UPDATE from setupChartWatcher)
+ * @param {Object}   actions  an array of chart actions to watch
+ * @param {function} callback  callback to execute when table is loaded.
+ */
+export function watchChartChanges(actions, callback) {
```

Comment:
There's no added value for this one.  It's exactly the same as `dispatchAddActionWatcher` since it will also create a uniqueID if not given.